src/librustc_data_structures/lib.rs

```diff
@@ -31,6 +31,13 @@ extern crate libc;
 #[macro_use]
 extern crate cfg_if;
 
+// Suppress unused crate lints:
+// when #[cfg(parallel_compiler)], there will be no uses
+// of these crates
+use rayon as _;
+use crossbeam_utils as _;
+use rayon_core as _;
```

Comment:
Does cargo support making these conditional in `Cargo.toml`?
Like
```
[target.'cfg(windows)'.dependencies]
winapi = { version = "0.3" }
```
, but for `cfg(parallel_compiler)`.

Reply:
AFAIK, Cargo features are the only way to conditionally enable dependencies.

Reply:
Ok, `parallel_compiler` is not even a Cargo feature, it's passed directly to `rustc` by rustbuild.